### DISCUSSION TITLE
Determining the status of other HomeKit accessories?

### DISCUSSION THREAD

**Discussion by helo-head** [HomeSpan/HomeSpan#407] [General]
After successfully creating two HomeSpan solutions that have been working flawlessly for over a year Im now hoping to build a solution similar to Brilliant's [controller](https://www.brilliant.tech/products/brilliant-control-single-switch-smart-lighting-smart-home-control?variant=white). Im replacing my DSC security system with a Konnected/Homebridge solution and want to use a M5Stack [Paper](https://shop.m5stack.com/products/m5paper-esp32-development-kit-comm-edition) solution to arm/disarm the system as well as show the current status of the security system. 

Ive read thru all the examples and believe example 16 covers the button part of my solution but I am not sure how to determine the status of the security system. Right now Im just using simulated lights triggered via a "when an accessory is controlled" automation but this seems cumbersome especially since I want to have 4 of these devices around the house. 

Is there any way in HomeSpan to query the status of other HomeKit accessories?

**Comment by HomeSpan** [comment]
I don't know of any way for one HomeKit device to query another separate HomeKit device via the HomeKit network.  That said, if you create two or more Accessories on the same device with HomeSpan, they can all communicate with each other from within the sketch (without needing to go through HomeKit).

One other option I'm just about to release is to use ESP-NOW for direct device-to-device communication.  I've completed and tested the code - just finishing up the documentation.   Of course this is specific to HomeSpan, not HomeKit, so would not work for communicating between HomeSpan and a commercial HomeKit device.

**Reply by helo-head** [reply]
Thanks, that's what I suspected. Unfortunately one of the issues with HomeKit is there is that it does provide any automation based on a change of state not driven by HomeKit such as someone turning on a light via a light switch. I guess my only option is to replace homebridge with my own solution using Konnected's API. Guess it's time to learn REST.
